humanmade/bulk-lighthouse
# GH-5 Add support for running tests in batches

GH-5 Add support for running tests in batches to avoid quota limits and requests failing due to too many requests being sent to PageSpeed API.

Note, there are some changes related to variable naming which seem to not be in the [main branch](https://github.com/humanmade/bulk-lighthouse/blob/4fc97f8ecc9ccf2906c050f136ddea153547bc17/cli.js#L74) but are in production via tag 1.0.2 as seen in this [commit](https://github.com/humanmade/bulk-lighthouse/commit/2e8dc8a60f8a1e7e4c727ca0874c6c12a20b23f2). Please let me know how to properly reconcile this issue if those variable naming changes are to be removed from this PR.

### Testing instructions

- Run the bulk-lighthouse script using the code in this PR and the tests should be run in batches provided your configuration file has a key `batchTests` set to `true`.
- Note you can also configure the number of URLs tested in each batch by adding the key `batchSize` to the config file with its value set to a number above 0. As mentioned in the issue description, this was done to give flexibility of configuration as the failure due to too many URLs being tested at once doesn't have a specific threshold so you can play around with this number to see what works for you or optionally just concurrently run all tests at once.
